Valigator.Newtonsoft.Json/ValigatorJsonReader.cs

```diff
@@ -48,8 +49,16 @@ public static Data<TDataValue> ReadValue<TDataValue>(JsonReader reader, JsonSeri
 			if (reader.TokenType == JsonToken.Null)
 				return SetValue(data, Option.None<TValue>());
 
-			if (serializer.Deserialize<TValue>(reader) is TValue value)
-				return SetValue(data, Option.Some(value));
+			try
+			{
+				if (serializer.Deserialize<TValue>(reader) is TValue value)
```

Comment:
`Deserialize<TValue>` throws a null ref if the value cannot be deserialized to TValue (ie: "NotAGuid" being deserialized into a Guid field). This method allows us to catch that exception and then set the error on the `Data` object.

Reply:
I've always struggled a bit on this one since you can't catch all errors of this kind (since not everything is Data<> like the root object), so catching these actually causes different types of deserialization errors to produce different results. Also, what happens here if the issue is badly formatted json (and it gets caught here) rather than a value that can't be converted.

Reply:
As far as I can tell, there isn't much of a way around this. If an exception is thrown here, it ends up trickling all the way out and the ModelBinding fails which results in a `ValidationProblemDetails` being created and the object we are deserializing into is lost. Finally, the default AspNetCore filters then get to that `ValidationProblemDetails` and it is then returned as a 500 and effectively skips over the ValigatorFilters

Reply:
I guess we could just let it fall through to `SetNull` as you originally coded it. However, we then lose the message which indicates what the improper value was.

Reply:
I guess at the end of the day, an `ExceptionFilter` can catch this and handle it. I think you convinced me. I'll look into yanking this out.

Reply:
I'll work on this tomorrow. My plan right now:
Throw a new exception type from the catch (`ValigatorJsonSerializationException`).
Create a `ValigatorJsonSerializationExceptionFilter` which can be easily added to the filters of projects that will turn a `ValigatorJsonSerializationException` into `ModelError[]` and will allow the use of a errorCreator function similar to the `AddValigator` call.